### PR TITLE
Correct join path and cascade for many-to-one relations

### PR DESCRIPTION
**Pull request by dstepanov**
Fixes https://github.com/micronaut-projects/micronaut-data/issues/939 https://github.com/micronaut-projects/micronaut-data/issues/815

### CODE REVIEW

File: data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java [hunk 1]
```diff
@@ -864,155 +869,211 @@ private void cascadeUpdate(Class<?> repositoryType, AnnotationMetadata annotatio
...
+        final RuntimePersistentProperty<?> associatedId = associatedEntity.getIdentity();
+        final BeanProperty associatedIdProperty = associatedId.getProperty();
+        // in the case of an one-to-one we ensure the inverse side set
+        if (child == null || persisted.contains(child)) {
+            return;
+        }
+        if (childAssociation.isForeignKey()) {
+            childAssociation.getInverseSide().ifPresent(inverse -> {
+                final BeanProperty property = inverse.getProperty();
+                property.set(child, parent);
```
Comment (graemerocher) [inline data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java:959]: What if the property is read-only? Maybe `withValue` and copy constructor approach as fallback?

File: data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java [hunk 2]
```diff
@@ -864,155 +869,211 @@ private void cascadeUpdate(Class<?> repositoryType, AnnotationMetadata annotatio
...
+                toPersist = new ArrayList<>(15);
+            }
+            for (Object o : entities) {
+                if (o == null || persisted.contains(o)) {
+                    continue;
+                }
+                if (inverse != null) {
+                    if (inverse.getKind() == Relation.Kind.MANY_TO_ONE) {
+                        final BeanProperty property = inverse.getProperty();
+                        property.set(o, parent);
```
Comment (graemerocher) [inline data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java:1016]: Same here what it is read-only like with Records?

**Review by graemerocher — APPROVED** [review]
Looks good, picked up a couple of cases where `set` is used and maybe we should support read-only entities like records (although that doesn't seem related to this change"

File: data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java (hunk 1 above)
Comment (dstepanov) [reply data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java:959]: I need to add more records tests to check all the scenarios.

**Pull request merged**